### PR TITLE
Update latest release tag to be 'stable' instead of 'latest'

### PR DESCRIPTION
**Pull request by j-wags**
We've had two problems with our `latest` tag on this repo:

1) I never updated it so it still points at 0.9.1 or something
2) ReadTheDocs uses `latest` as a special keyword to mean "latest commit on the main development branch", and by making a `latest` tag in this repo we overrode that, so now our "bleeding edge" docs aren't from the right place. 

- [x] Make a new tag called `stable`
- [x] Add "run the `stable` tag-updating action" to the [standard release process](https://github.com/openforcefield/openff-toolkit/issues/286)
- [x] Ensure that binder examples build+run
- [x] Ensure that docs render for the `stable` tag
- [x] Delete the `latest` tag
- [x] Ensure that ReadTheDocs'  "latest" docs are really built from the last commit

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,6 +9,6 @@ jobs:
     - name: Update latest tag
```
Comment (Yoshanuikabundi) [inline .github/workflows/release.yml:9]:
```suggestion
    - name: Update stable tag
```

Comment (Yoshanuikabundi) [reply .github/workflows/release.yml:9]: The name of the entire workflow (first line) should be updated too.

**Review by Yoshanuikabundi — COMMENTED** [review]
This looks good! There are a bunch of links in the notebooks themselves that point to the latest version on RTD that should now point to stable:

- examples/conformer_energies/conformer_energies.ipynb
- examples/toolkit_showcase/toolkit_showcase.ipynb
- examples/SMIRNOFF_simulation/run_simulation.ipynb
- examples/using_smirnoff_in_amber_or_gromacs/convert_to_amber_gromacs.ipynb

**Review by Yoshanuikabundi — APPROVED** [review]
Poifect

**Comment by j-wags** [comment]
Thanks! I'll merge this when tests pass.

**Pull request merged**